### DISCUSSION TITLE
Images copy/move with adb does not show up in folder

### DISCUSSION THREAD

**Discussion by GHolk** [deckerst/aves#591] [Q&A] [answered]
Hey, I am new to Aves.
I copy some images from pc to android device with adb and some similar tools,
but the images does not show up.
(similar tool: [SSHelper](https://arachnoid.com/android/SSHelper/index.html))

I make some search and believe this issue is related to MediaStore API.
I find a app [SD Scanner] on f-droid can make Aves see these images.
(And a adb `am` command which does not work.)

[SD Scanner]: https://f-droid.org/zh_Hant/packages/com.gmail.jerickson314.sdscanner/

Is it possible to fix this issue? There are some issues about mediastore but I have not look into them.

**Comment by deckerst** [comment, marked as answer]
> I make some search and believe this issue is related to MediaStore API.

Yes, it's related. When an app or tool move media files on Android storage, it should call the right APIs so that Media Store is aware of them. Otherwise, apps that rely on the Media Store (like Aves) won't list them as media.

> Is it possible to fix this issue?

The issue is on the side of these apps and tools that move media without letting Media Store know about it. Tools that force a rescan (like SD Scanner) are useful but they're a band-aid after the fact.

**Reply by deckerst** [reply]
By the way, I believe that when you connect an Android device to typical Windows or Linux and transfer files via the Explorer/Files desktop apps, it correctly follows MTP (Media Transfer Protocol) so that media files are registered in the Media Store.

**Reply by GHolk** [reply]
Thank to your explaination!

I prefer wireless connection so I don't use MTP.
It seem that most wlan solution does not support Media Store :'(

**Reply by e-t-l** [reply]
MTP isn't involved if the file transfers don't use adb push/pull. I made the mistake of reorganizing some photos in different folders using adb shell and the bash `mv` command. Now the photos are not showing up in Aves. I tried the command `adb shell content call --method scan_volume --uri content://media --arg external_primary` you suggested [here](https://github.com/deckerst/aves/discussions/1812#discussioncomment-16687124), and while the images no longer show up in Aves with their old paths (displaying as corrupted thumbnails), they now don't show up at all :/

**Reply by deckerst** [reply]
@e-t-l you could try to reboot your device and wait for Android to find your media.
